substrate/frame/society/src/lib.rs

```diff
@@ -1488,13 +1503,58 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		let rotation_period = voting_period + claim_period;
 		let now = T::BlockNumberProvider::current_block_number();
 		let phase = now % rotation_period;
-		if phase < voting_period {
+		if now > Self::next_intake_at() {
+			Period::IntakeDelay { elapsed: now - Self::next_intake_at() }
```

Comment:
if delayed for too long we wanna make sure we do not inter the `Voting` period again